KtlintRuleException thrown after updating Android Studio

After updating Android Studio to Android Studio Otter 3 Feature Drop | 2025.2.3, I started getting these errors non-stop:


> KtLintRuleException
An error occurred in a rule. Please see stacktrace below for rule that caused the problem and contact maintainer of the rule when the error can be reproduced. com.pinterest.ktlint.rule.engine.api.KtLintRuleException: Rule 'compose:composable-annotation-naming' throws exception in file ' ' at position (1:1) Rule maintainer: Compose Rules Issue tracker : https://github.com/mrmans0n/compose-rules/issues Repository : https://github.com/mrmans0n/compose-rules at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRule(RuleExecutionContext.kt:75) at com.pinterest.ktlint.rule.engine.internal.CodeFormatter.executeRule(CodeFormatter.kt:138) at com.pinterest.ktlint.rule.engine.internal.CodeFormatter.format(CodeFormatter.kt:109) at com.pinterest.ktlint.rule.engine.internal.CodeFormatter.format(CodeFormatter.kt:58) at com.pinterest.ktlint.rule.engine.internal.CodeFormatter.format(CodeFormatter.kt:28) at com.pinterest.ktlint.rule.engine.api.KtLintRuleEngine.lint(KtLintRuleEngine.kt:91) at com.nbadal.ktlint.KtlintRuleEngineWrapper.executeKtlint(KtlintRuleEngineWrapper.kt:169) at com.nbadal.ktlint.KtlintRuleEngineWrapper.lint(KtlintRuleEngineWrapper.kt:63) at com.nbadal.ktlint.KtlintAnnotator.collectInformation(KtlintAnnotator.kt:69) at com.nbadal.ktlint.KtlintAnnotator.collectInformation(KtlintAnnotator.kt:24) at com.intellij.codeInsight.daemon.impl.ExternalToolPass.collectInformationWithProgress(ExternalToolPass.java:136) at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:92) at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:71) at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:435) at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.use(trace.kt:29) at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$3(PassExecutorService.java:431) at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.tryRunReadAction(NestedLocksThreadingSupport.kt:826) at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1221) at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$4(PassExecutorService.java:421) at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:681) at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:756) at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:712) at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:680) at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78) at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:420) at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:395) at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.cacheFileTypesInside(FileTypeManagerImpl.java:852) at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$1(PassExecutorService.java:395) at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:258) at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:393) at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:265) at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source) at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source) at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) Caused by: java.lang.AbstractMethodError: Method io/nlopez/compose/rules/ktlint/ComposableAnnotationNamingCheck.beforeVisitChildNodes(Lshadow/org/jetbrains/kotlin/com/intellij/lang/ASTNode;Lkotlin/jvm/functions/Function3;)V is abstract at io.nlopez.compose.rules.ktlint.ComposableAnnotationNamingCheck.beforeVisitChildNodes(ComposableAnnotationNamingCheck.kt) at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleWithAutocorrectApproveHandlerOnNodeRecursively(RuleExecutionContext.kt:173) at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:99) at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRule(RuleExecutionContext.kt:72) ... 35 more


It seems that it happens when I alternate between files. This wasn't happening before the update. 

<img width="368" height="97" alt="Image" src="https://github.com/user-attachments/assets/e14ba4a3-e15a-4d16-8b5b-4b66ca9dbc81" />

The dependencies I am using are compatible according to the documentation:

ktlint = "14.0.1"
ktlintCompose = "0.5.3"
Ktlint Ruleset Version: 1.8.0
Ktlint plugin version: 0.30.2
Kotlin: 2.3.0

I am guessing the Android Studio update changed something and it broke the plugin? Because that's the only change I made for that to start happening. A colleague is having the same issue using the same Android Studio version.

--- a/rules/ktlint/build.gradle.kts
+++ b/rules/ktlint/build.gradle.kts
@@ -18,11 +18,12 @@ if (!project.hasProperty("uberJar")) {
 
 tasks.shadowJar {
     // Relocate packages that may conflict with the ones IntelliJ IDEA provides as well.
-    // See https://github.com/nbadal/ktlint-intellij-plugin/blob/main/lib/build.gradle.kts
+    // See https://github.com/nbadal/ktlint-intellij-plugin/blob/main/ktlint-lib/build.gradle.kts
     relocate("org.jetbrains.concurrency", "shadow.org.jetbrains.concurrency")
     relocate("org.jetbrains.kotlin.psi.KtPsiFactory", "shadow.org.jetbrains.kotlin.psi.KtPsiFactory")
     relocate("org.jetbrains.kotlin.psi.psiUtil", "shadow.org.jetbrains.kotlin.psi.psiUtil")
     relocate("org.jetbrains.org", "shadow.org.jetbrains.org")
+    relocate("org.jetbrains.kotlin", "shadow.org.jetbrains.kotlin")
 }
 
 dependencies {

